### ISSUE TITLE
additive_terms vs extra_regressors_additive

### ISSUE THREAD

**Issue by Shafi2016** [facebook/prophet#1799]
Hello Ben Letham,

Thanks for the nice work!. I am trying to understand a few points. This is the first time I started applying the prophet model. It has an amazing performance. I found that for the best forecast, **additive_terms and extra_regressors_additive** become almost identical.  For the bad forecast, there is some large difference bw the two terms. I do not know the institution behind this. So additive_terms  : y(t) = trend(t) + seasonality(t) + beta * regressor(t).  Thanks.

**Comment by bletham** [comment]
The `additive_terms` is just going to be the combination of everything adding to the trend. So, with everything additive like `y(t) = trend(t) + seasonality(t) + beta * regressor(t)`, the `extra_regressors_additive` corresponds to `beta * regressor(t)` and `additive_terms` corresponds to `seasonality(t) + beta * regressor(t)`. If these are the same, it means the seasonality was fit to be basically 0. They will be different when there is significant seasonality. If you look at the components plot with `m.plot_components(fcst)` you should be able to see it there.

As for why it would fit seasonality in one forecast vs. the other, if you post the plots of the two time series with `m.plot` I may be able to offer some thoughts.

**Comment by Shafi2016** [comment]
Thanks a lot, @bletham. I have one target with one predictor using monthly data. I have taken 4 Lag of predictors and rolling averages based on lags. Here are plots of additive term VS extra_regressors_additive. Their pattern match quite well with the target variables. 
![image](https://user-images.githubusercontent.com/56795978/107125192-59fb4080-6876-11eb-8385-e956c5bd2dcb.png)

![image](https://user-images.githubusercontent.com/56795978/107125170-3e903580-6876-11eb-9cf9-dfe8e8e01550.png)

**Comment by bletham** [comment]
For data snooping / performance with lagged regressors: I think the only way that would happen would be if you are using test-set values for computing the future values of the lagged regressors. Let me explain:
Suppose you are forecasting y(t) and you include z(t) = y(t-7) as a regressor that is y lagged by 7 days. You have data from t=1 to T.
Suppose you have a test set of dates T+1, ..., T+h. To make predictions for this, you have to fill in the value for the regressors z(T+1), ..., z(T+h), which of course equals y(T - 6), ..., y(T+h - 7). If h is greater than 7, then this is a problem because these are test set y's, and that would be leaking test labels to the model. If h is <= 7, then you can compute the regressor values solely from the train set and all is well.

A corollary of this is that lagged regressors are only useful if the desired prediction horizon is less than the shortest lag in the regressors. See also #1802 for some ongoing discussion on this.

**Comment by Shafi2016** [comment]
@bletham Thanks a lot! In reality, I do not use lags. First, I want to forecast, 4 months into the future. So I take 4 lags of independent variables (x) and dependent variable(y) as AR. Then I take rolling averages of these lags. I have given below the future data preparation codes by using timetk  R package. Then I divide this data into train and test keeping in the view time sequence of course. Then we have a future forecast where there is no value for y. Actually adding lags does not improve a rather it is the rolling average of lags that improves but this is limited to the prophet model only. But the future forecast is a just straight line only even though it matches perfectly on the test data. Please shed a light on it.
```
horizon    <- 4
lag_period <- 4
rolling_periods <- c(4: 8)

 data_pre <- data %>%
        bind_rows(
        future_frame(.data = .,.date_var = date, .length_out = horizon)
        ) %>%     
                                                                       
     # add lags
   tk_augment_lags(
      .value = y: X, 
      .lags = lag_period)  %>%
      
    tk_augment_slidify(
      .value  =  y_lag4 : X_lag4,
     .period  = rolling_periods,
      .f       = mean,
      .align   = "center", 
      .partial = TRUE)  
```

**Comment by bletham** [comment]
I'm not familiar with timetk and so can't comment on what this particular code is doing, but the safest way to be sure would be to not put your test data anywhere in your training loop. I don't see any filtering on dates here to make sure that none of the test-set y's are making it into the training data. Note that you need to filter y's before the test-set cutoff date _before_ computing any of the regressors.

**Comment by Shafi2016** [comment]
Thank you so much. I checked manually both training and test data sets. I do not see any problem though.
